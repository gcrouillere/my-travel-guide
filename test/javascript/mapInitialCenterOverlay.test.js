import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer';

import MapInitialCenterOverlay from '../../app/javascript/packs/articleForm/contentMenu/mapInitialCenterOverlay'

configure({ adapter: new Adapter() });

describe('MapInitialCenterOverlay test suite', () => {

  let abandonMapCreation, addNewMap
  abandonMapCreation = addNewMap = jest.fn()

  const mapInitialCenterOverlayTree = renderer
    .create(<MapInitialCenterOverlay abandonMapCreation={abandonMapCreation} addNewMap={addNewMap}/>)
    .toJSON()
  expect(mapInitialCenterOverlayTree).toMatchSnapshot();

  const wrapper = mount(<MapInitialCenterOverlay abandonMapCreation={abandonMapCreation} addNewMap={addNewMap}/>)

  it('passes location to upper component', () => {
    wrapper.instance().handleMap()
    expect(addNewMap).toHaveBeenCalled()
  })

  it('passes overlay closing to upper component', () => {
    wrapper.find(".close").simulate('click')
    expect(abandonMapCreation).toHaveBeenCalled()
  })

})
