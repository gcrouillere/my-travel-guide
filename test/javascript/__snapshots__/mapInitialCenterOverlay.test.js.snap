// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="mapInitialCenterOverlay "
>
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  <p
    className="mapOverlayTitle"
  >
    Which area do you want to show ?
  </p>
  <p
    className="mapOverlaydescription"
  >
    If no suggestion is given, try a near match. You 'll be able to fine tune the map in the article editor.
  </p>
  <div
    className="mapLocation"
  >
    <div
      className="locationInput input-group mb-3"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-default"
        >
          Map's center
        </span>
      </div>
      <input
        aria-describedby="inputGroup-sizing-default"
        aria-label="Default"
        className="form-control mapLocationInput"
        draggable={true}
        id="mapLocationmenu"
        onChange={[Function]}
        onDragStart={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
